src/idlcxx/src/streamers.c

```diff
@@ -106,15 +114,172 @@ process_instance(
+    || putf(&streams->move, wfmt, depth, accessor, maximum, "move")
+    || putf(&streams->max, mfmt, depth, maximum))
+    return IDL_RETCODE_NO_MEMORY;
+
+  if (is_key)
+  {
+    if (putf(&streams->key_read, rfmt, depth, accessor)
+      || putf(&streams->key_write, wfmt, depth, accessor, maximum, "write")
+      || putf(&streams->key_move, wfmt, depth, accessor, maximum, "move")
+      || putf(&streams->key_max, mfmt, depth, maximum))
+    return IDL_RETCODE_NO_MEMORY;
+  }
+
+  char* new_accessor = NULL;
+  if (idl_asprintf(&new_accessor, "%s[i_%u]", accessor, depth) < 0)
+    return IDL_RETCODE_NO_MEMORY;
+
+  idl_retcode_t ret = IDL_RETCODE_OK;
+  if (idl_is_sequence(seq->type_spec))
+    ret = unroll_sequence (pstate, streams, (idl_sequence_t*)seq->type_spec, depth + 1, new_accessor, is_key);
+  else
+    ret = check_endpoint (pstate, streams, seq->type_spec, new_accessor, is_key);
+
+  if (ret != IDL_RETCODE_OK)
+    goto done;
+
+  //close sequence
+  wfmt = "  } //i_%1$u\n";
+  rfmt = maximum ? "  } //i_%1$u\n  if (se_%1$u > %2$u) %3$s.resize(%2$u);\n" : wfmt;
+  mfmt = maximum ? wfmt : "  } //i_%1$u\n  str.position(SIZE_MAX);\n";
+
+  if (putf(&streams->read, rfmt, depth, maximum, accessor)
+    || putf(&streams->write, wfmt, depth)
+    || putf(&streams->move, wfmt, depth)
+    || putf(&streams->max, mfmt, depth))
+  return IDL_RETCODE_NO_MEMORY;
+  if (is_key)
+  {
+    if (putf(&streams->key_read, rfmt, depth, maximum, accessor)
+      || putf(&streams->key_write, wfmt, depth)
+      || putf(&streams->key_move, wfmt, depth)
+      || putf(&streams->key_max, mfmt, depth))
+      return IDL_RETCODE_NO_MEMORY;
+  }
+
+done:
+  free(new_accessor);
+
+  return ret;
+}
+
+static idl_retcode_t
+process_instance(
+  const idl_pstate_t *pstate,
+  struct streams *streams,
+  const idl_declarator_t* declarator,
+  const idl_type_spec_t* type_spec,
+  bool is_key)
+{
+  const char *name = get_cpp11_name(declarator);
+  char *accessor = NULL;
+
+  if (idl_asprintf(&accessor, "instance.%s()", name) < 0)
+    return IDL_RETCODE_NO_MEMORY;
+
+  //unroll arrays
+  const char* afmt = "  for (uint32_t _a = 0; _a < %1$u; _a++) {\n";
+  if (idl_is_array(declarator))
+  {
+    uint32_t entries = ((const idl_literal_t*)declarator->const_expr)->value.uint32;
+    if (putf(&streams->write, afmt, entries)
+      || putf(&streams->read, afmt, entries)
+      || putf(&streams->move, afmt, entries)
+      || putf(&streams->max, afmt, entries))
+      return IDL_RETCODE_NO_MEMORY;
+    if (is_key)
+    {
+      if (putf(&streams->key_write, afmt, entries)
+        || putf(&streams->key_read, afmt, entries)
+        || putf(&streams->key_move, afmt, entries)
+        || putf(&streams->key_max, afmt, entries))
+        return IDL_RETCODE_NO_MEMORY;
+    }
+
+    //update accessor: accessor = accessor + "[_a]"
+    char *tmp = NULL;
+    if (idl_asprintf(&tmp, "%s[_a]", accessor) < 0)
+      goto err;
```

Comment:
Please use IDL_PRINTA, it was specifically introduced to avoid having to do `alloc` + `copy` + `free` and having to include that into the logic throughout the generator.

Reply:
will do